packages/auth/src/authClient.ts

```diff
@@ -438,7 +439,7 @@ export class AuthClient {
   public async loadAuthSessionState(
     state: string,
   ): Promise<AuthSession | undefined> {
-    const parsedState = JSON.parse(state);
+    const parsedState: AuthSessionState = JSON.parse(state);
```

Comment:
it's not safe enough to just assume that the user is going to pass in a string that is the correct state. Instead of doing a blind JSON.parse and cast to `AuthSessionState`, we should add an additional method to validate that the state passed in can be parsed to a valid `AuthSessionState` and error if the srting is invalid.

Reply:
added a `deserializeState` function. should we log the parsed state in the error if it can successfully be parsed as JSON but isn't a valid `AuthSessionState` object?